### DISCUSSION TITLE
Define fork deployers update strategy

### DISCUSSION THREAD

**Discussion by dianjuar** [angular-schule/ngx-deploy-starter#18] [General]
Hello.

I think that is useful to have a guide to know how to update the new releases, features, bug fixes that are being released, to have them in our forks.

It would be nice to adopt a strategy that allows us to have our forks updated instead of making all the changes by hand, copying and pasting the new code. I think that rebase our changes to the master branch is a good strategy and will avoid the copy and paste.

--------

I have to update my fork to adopt the new releases I will go with the rebase strategy to set it down and document my experience here

**Comment by JohannesHoppe** [comment]
👍

**Comment by dianjuar** [comment]
The rebase strategy was a mess, no good experience doing it, a lot of conflicts, and definitely not something nice to do.

I already tried doing a cheery-pick of the changes that I wanted no implement on my fork and was easy, fun, and straightforward. 
I did it on implementing the [`--no-build`](https://github.com/bikecoders/ngx-deploy-npm/pull/31) option and was very easy. It even added @JohannesHoppe to the co-authors of the commit 😄. 

In the next weeks, I will implement all the RFC proposals using cherry-pick, I will let you know.

**Comment by JohannesHoppe** [comment]
I usually do such a job by comparing the folders 1:1 with a tool, eg. Beyond Compare.
Does that work for you?

For the future I will try to handle version bumps in a single PR, so that it is less work for you.

**Reply by dianjuar** [reply]
> For the future I will try to handle version bumps in a single PR, so that it is less work for you.

This would help a lot!

**Comment by dianjuar** [comment]
For me, this can be closed, I'm really happy with the results of cherry-picking, easy, straight forward and on the commits picked leaves the co-author that personally I find it really nice.

I will put my strategy followed:

1. Select the feature/fix that you want to implement in your fork
2. Search for the commits involved in that feature 
   - You may find that several commits were needed to implement the feature that you want, some fixes next to the implementation for example
3. Create a branch
4. Do `git cherry-pick COMMIT_HASH`
   - Resolve conflicts and change the things that may apply to your particular case
   - Do the same for all the commits involved
5. Create the PR with the feature/fix 🎉 

---

These are the features/fixes implemented at the moment of writing this on the fork [ngx-deploy-npm](https://github.com/bikecoders/ngx-deploy-npm) (more coming) 

- feat: Add no build option [ngx-deploy-npm#31](https://github.com/bikecoders/ngx-deploy-npm/pull/31)
- fix: add options to ngAdd schematic [ngx-deploy-npm#43](https://github.com/bikecoders/ngx-deploy-npm/pull/43)
- fix: throws an error if app building fails [ngx-deploy-npm#48](https://github.com/bikecoders/ngx-deploy-npm/pull/48)
